fuel/src/main/kotlin/com/github/kittinunf/fuel/util/Base64.kt

```diff
@@ -0,0 +1,80 @@
+/*
+Copied From: https://gist.github.com/hrules6872/e2d4d02a1e8d3c6328ae5aeabc430b96
+ */
+package com.github.kittinunf.fuel.util
+
+import java.io.ByteArrayOutputStream
+
+fun String.encodeBase64ToString(): String = String(this.toByteArray().encodeBase64())
+fun String.encodeBase64ToByteArray(): ByteArray = this.toByteArray().encodeBase64()
+fun ByteArray.encodeBase64ToString(): String = String(this.encodeBase64())
+
+fun String.decodeBase64(): String = String(this.toByteArray().decodeBase64())
+fun String.decodeBase64ToByteArray(): ByteArray = this.toByteArray().decodeBase64()
+fun ByteArray.decodeBase64ToString(): String = String(this.decodeBase64())
+
+fun ByteArray.encodeBase64(): ByteArray {
+    val table = (CharRange('A', 'Z') + CharRange('a', 'z') + CharRange('0', '9') + '+' + '/').toCharArray()
+    val output = ByteArrayOutputStream()
+    var padding = 0
+    var position = 0
+    while (position < this.size) {
+        var byte = this[position].toInt() and 0xFF shl 16 and 0xFFFFFF
+        byte = if (position + 1 < this.size) byte or (this[position + 1].toInt() and 0xFF shl 8) else padding++
+        if (position + 2 < this.size) byte = byte or (this[position + 2].toInt() and 0xFF) else padding++
+        for (i in 0 until 4 - padding) {
+            val c = byte and 0xFC0000 shr 18
+            output.write(table[c].toInt())
+            byte = byte shl 6
+        }
+        position += 3
+    }
+    for (i in 0 until padding) {
+        output.write('='.toInt())
+    }
+    return output.toByteArray()
+}
+
+fun ByteArray.decodeBase64(): ByteArray {
+    val table = intArrayOf(-1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
```

Comment:
would this table not be better in a companion object or declared as a const val, as everytime this function is called it will have to declare this array to then junk it?

Reply:
I can't do const val because int[] is primitive types. 
I had 

```kotlin
val decodedTable = intArrayOf()
``` 
outside of the function

Inside of the decodeBase64 function, 
I had
```kotlin
 val table = decodedTable
```